packages/components/pie-textarea/src/index.ts

```diff
@@ -19,25 +22,48 @@ export class PieTextarea extends RtlMixin(LitElement) implements TextareaProps {
     static shadowRootOptions = { ...LitElement.shadowRootOptions, delegatesFocus: true };
 
     @property({ type: Boolean, reflect: true })
-    public disabled?: TextareaProps['disabled'];
+    public disabled = defaultProps.disabled;
 
     @property({ type: String })
     @validPropertyValues(componentSelector, sizes, defaultProps.size)
-    public size?: TextareaProps['size'] = defaultProps.size;
+    public size = defaultProps.size;
+
+    @property({ type: String })
+    @validPropertyValues(componentSelector, resizeModes, defaultProps.resize)
+    public resize = defaultProps.resize;
+
+    @query('textarea')
+    private _textarea!: HTMLTextAreaElement;
+
+    private handleResize () {
+        if (this.resize === 'auto') {
+            this._textarea.style.height = 'auto';
+            this._textarea.style.height = `${this._textarea.scrollHeight + 2}px`; // +2 for border thicknesses
+        }
+    }
+
+    updated (changedProperties: PropertyValues<this>) {
+        if (this.resize === 'auto' && (changedProperties.has('resize') || changedProperties.has('size'))) {
+            this.handleResize();
+        }
+    }
 
     render () {
         const {
             disabled,
+            resize,
             size,
         } = this;
 
         return html`
             <div
-                class="c-textarea"
-                data-test-id="pie-textarea-shell"
-                data-pie-size=${ifDefined(size)}>
+                class="c-textareaWrapper"
+                data-test-id="pie-textarea-wrapper"
+                data-pie-size="${size}"
```

Comment:
i think we usually just assign the prop name to the element such as size and resize. 
has this changed?

Reply:
Do you mean `size="${size}"` instead? I think @ashleynolan was saying that technically this is invalid HTML because `size` is not a valid HTML attribute for the `textarea` element, so we want to avoid potentially being punished in terms of SEO if they start checking the HTML inside the shadow DOM.

Reply:
just wondering, does it make sense to change `size` props to `data-pie-size` inside `pie-modal` and `pie-spinner` components:

https://github.com/justeattakeaway/pie/blob/96ea944afa3526717ca253f66057823afa95839d/packages/components/pie-spinner/src/index.ts#L44
https://github.com/justeattakeaway/pie/blob/96ea944afa3526717ca253f66057823afa95839d/packages/components/pie-modal/src/index.ts#L404

Reply:
Yes, I think so. Shall I add it as an AC to the ticket for aligning props and types?

Reply:
yes, I think DSW-2191 is fine
thank you